File: webshop-backend/src/main/java/ee/maarja/webshopbackend/entity/Product.java
package ee.maarja.webshopbackend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private double price;

    @Column(length = 1855)
    private String description;
    //private String category;
    private String image;

    @ManyToOne
    private Category category;

    @OneToOne(cascade = CascadeType.ALL)
    private Rating rating;
}

File: webshop/src/models/Product.ts
import { Category } from "./Category"

export class Product {
  constructor(
    public title: string = "",
    public price: number = 0,
    public description: string = "",
    public category: Category = {id: 0, name: ""},
    public image: string = "",
    public rating: Rating = {rate: 0, count: 0},
    public id?: number,
  ) {}
}

type Rating = {
  rate: number
  count: number
}